cmd/queue.go

```diff
@@ -0,0 +1,404 @@
+func checkEventDeliveryinZSET(a *app) *cobra.Command {
+	var id string
+	var idType string
+	cmd := &cobra.Command{
+		Use:   "checkzset",
+		Short: "Event delivery in ZSET",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cfg, err := config.Get()
+			if err != nil {
+				return err
+			}
+			if cfg.Queue.Type != config.RedisQueueProvider {
+				log.Fatalf("Queue type error: Command is available for redis queue only.")
+			}
+			if util.IsStringEmpty(id) {
+				return errors.New("please provide an eventDelivery ID or equivalent taskq.Message ID")
+			}
+			ctx := context.Background()
+			q := a.eventQueue.(*redisqueue.RedisQueue)
+			bodies, err := q.ZRangebyScore(ctx, "-inf", "+inf")
+			if err != nil {
+				return err
+			}
+
+			inZSET := false
+
+			var msg taskq.Message
+			for _, body := range bodies {
+				err := msg.UnmarshalBinary([]byte(body))
+				if err != nil {
+					return err
+				}
+				switch idType {
+				case "eventdev":
+					value := string(msg.ArgsBin[argsSlice:])
+					if value == id {
+						inZSET = true
+					}
+				case "msg":
+					if msg.ID == id {
+						inZSET = true
+					}
+				}
+			}
+			if inZSET {
+				log.Printf("Event ID: %v in inZSET: True", id)
+			} else {
+				log.Printf("Event ID: %v in inZSET: False", id)
+			}
+			return nil
+		},
+	}
+	cmd.Flags().StringVar(&id, "id", "", "eventDelivery ID or taskq.Message ID")
+	cmd.Flags().StringVar(&idType, "type", "eventdev", "eventdev or msg")
+	return cmd
+}
+
+//Check if eventDelivery is in pending (stream)
+func checkEventDeliveryinPending(a *app) *cobra.Command {
+	var id string
+	var idType string
+	cmd := &cobra.Command{
+		Use:   "checkpending",
+		Short: "Event delivery on pending",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cfg, err := config.Get()
+			if err != nil {
+				return err
+			}
+			if cfg.Queue.Type != config.RedisQueueProvider {
+				log.Fatalf("Queue type error: Command is available for redis queue only.")
+			}
+			if util.IsStringEmpty(id) {
+				return errors.New("please provide an eventDelivery Id or taskq.Message ID")
+			}
+			ctx := context.Background()
+			q := a.eventQueue.(*redisqueue.RedisQueue)
+			pending, err := q.XPendingExt(ctx, "-", "+")
+			if err != nil {
+				log.Printf("Error fetching Pending: %v", err)
+			}
+
+			inPending := false
+
+			var msg *taskq.Message
+			for i := range pending {
+				xmsgInfo := &pending[i]
+				id := xmsgInfo.ID
+
+				xmsgs, err := q.XRangeN(ctx, id, id, 1).Result()
+				if err != nil {
+					return err
+				}
+
+				if len(xmsgs) != 1 {
+					log.Printf("redisq: can't find pending message id=%q", id)
+				}
+				err = unmarshalMessage(msg, &xmsgs[0])
+				if err != nil {
+					return err
+				}
+				switch idType {
+				case "eventdev":
+					value := string(msg.ArgsBin[argsSlice:])
+					if value == id {
+						inPending = true
+					}
+				case "msg":
+					if msg.ID == id {
+						inPending = true
+					}
+				}
+
+			}
```

Comment:
is it possible to have this logic in a separate func?

Reply:
the check in the Switch statement?

Reply:
Based on our conversation earlier, I've done some clean up to help make those components more reusable. Still working on a delete function.